server/models/ticket.js

```diff
@@ -40,7 +67,43 @@ TicketSchema.statics = {
         reject(error);
       });
     });
+  },
+
+  addAction(id, action, timestamp, userId) {
+    const builtActionObject = {
+      actionTaken: action,
+      user: userId,
+      timestamp: timestamp || new Date.getTime()
+    };
+
+    return new Promise((resolve, reject) => {
+      this.findByIdAndUpdate(id, { $push: { actions: builtActionObject } }, { new: true }, (err, ticket) => {
+        if (ticket) {
+          return resolve(ticket);
+        }
+        reject(err);
+      });
+    });
+  },
+
+  removeAction(id, action, timestamp, userId) {
+    const builtActionObject = {
+      actionTaken: action,
+      user: userId,
+      timestamp: timestamp || new Date.getTime()
```

Comment:
It is milliseconds. We will also have the exactly details of any action we want to delete, so it will match correctly.